AnimateQuotes.js

```diff
@@ -1,21 +1,20 @@
 import React, { useRef, useEffect, useState } from 'react';
 import { Animated, Text, View } from 'react-native';
-import { useIsShake } from './useIsShake';
-import { QUOTES_IN_DURATION, QUOTES_IN_DELAY, QUOTES_OUT_DURATION, SVG_IN_DELAY, SVG_IN_DURATION, SVG_OUT_DURATION, TRANSITION_TIME_2 } from './constants';
+import { QUOTES_IN_DURATION, QUOTES_IN_DELAY, QUOTES_OUT_DURATION, quotes } from './constants';
+import { getRandomInt } from './utils';
+import Quotes from "./Quotes";
 
 export default function AnimateQuotes(props) {
-  const [isRegistered, setIsRegistered] = useState(false);
-  const [hasToggledBefore, setHasToggledBefore] = useState(false);
+  const [isQuoteAnimationRunning, setIsQuoteAnimationRunning] = useState(false);
+  const [quote, setQuote] = useState(0);
 
   const fadeAnim = useRef(new Animated.Value(0)).current;
   const sizeAnim = useRef(new Animated.Value(0.5)).current;
 
-  const { toggle } = useIsShake();
+  const { isShakeTriggered, setIsToggleReady } = props.useIsShake;
+  const { hasToggledBefore, setHasToggledBefore } = props.toggledBefore;
 
-  // lands on Quotes after LogoSvg fades out first time
-  if (toggle && !hasToggledBefore && !isRegistered) {
-    setIsRegistered(true);
-    setHasToggledBefore(true);
+  function quoteAnimationIn() {
```

Comment:
Composed animations moved into functions to be called in conditional logic in order to make code more legible.